pkg/apis/apps/validation/validation.go

```diff
@@ -572,7 +577,7 @@ func ValidateDeploymentSpec(spec *apps.DeploymentSpec, fldPath *field.Path, opts
 	if err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("selector"), spec.Selector, "invalid label selector"))
 	} else {
-		allErrs = append(allErrs, ValidatePodTemplateSpecForReplicaSet(&spec.Template, selector, spec.Replicas, fldPath.Child("template"), opts)...)
+		allErrs = append(allErrs, ValidatePodTemplateSpecForReplicaSet(&spec.Template, nil, selector, spec.Replicas, fldPath.Child("template"), opts)...)
```

Comment:
This function is called from ValidateDeployment and ValidateDeploymentUpdate - you need to pass the argument down from those.

Reply:
It seems that there is also a ValidateReplicaSet :)